tools/pkg/engine_build_configs/lib/src/build_config_runner.dart

```diff
@@ -579,6 +579,27 @@ final class BuildRunner extends Runner {
     return min(multiplier * cores, 1000);
   }();
 
+  static final RegExp _gccRegex =
```

Comment:
All messages aren't routed through the `eventHandler` today.  Your proposal would require that to be the case.  I would expect printing proper paths would be fundamental to BuildRunner and not something every client of BuildRunner should set up.

Reply:
> All messages aren't routed through the `eventHandler` today.

What's missing? Looking below, `_ninjaProgress` returns `true` only after it sends a line of output to the event handler, and when it returns `false`, there's an early return that also skips the path rewriting.

Reply:
> printing proper paths would be fundamental

This library doesn't really "print" anything. That's the thinking behind the event handling part of the API. Clients of this API get to have an opinion about what the output seen by users looks like. If there's some data that clients need that isn't getting sent to the event handler, then we should fix that.

Reply:
> What's missing? Looking below, _ninjaProgress returns true only after it sends a line of output to the event handler, and when it returns false, there's an early return that also skips the path rewriting.

_ninjaProgress is parsing the output from ninja (like this PR does) and selectively forwards it on to the event handler.  In our case it doesn't forward it on since it isn't a progress message.  That's why modifying the line had to happen in BuildRunner.

> Clients of this API get to have an opinion about what the output seen by users looks like.

Today they don't, the output just gets forward to stdout without their opinion.  I suspect you think everything that is getting printed out is going through `et` which isn't the case.  Given the choice to print wrong paths or correct paths we don't have anyone that wants the wrong paths so I don't think it's worth making that optional or something where a user should express their opinion.  In my view this is a bug and not a missing extension point where users can change the behavior.

Reply:
Oops. Yeah, you're right. I misread the code, and I see what you're saying now.

Here is my *non-blocking* advice: Similar to how progress information from ninja is parsed out and handed to the client as a `RunnerProgress` event, add a function like `_ninjaError` or `_compilerError` that parses out compiler messages (errors/warnings/notes) and hands them to the client with a `RunnerError` event. This way `et` will have the option of collecting/displaying compiler messages however it likes. Other use cases might be adding color/emoji/ascii art, writing to a json file, etc.

Reply:
So I agree with Zach's (ultimate) position, but also agree it should be non-blocking - this is a strict improvement, is tested, and is small in scope. The most I would ask is a TODO/GH issue tracking _improving_ this by uplifting it as a `_xError`, but not as a blocker for landing this (positive) change.

Reply:
done